### PR TITLE
Support for fund and quote_fund for wallet funding

### PR DESCRIPTION
**Pull request by rohan-agarwal-coinbase**
### What changed? Why?
-  Creates new crypto_amount and fiat_amount classes to wrap the underlying models, as well as fund_operation and fund_quote.
- Updates wallet_address and wallet for new fund and quote_fund functions.

### Testing
- Unit tests

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: cdp/fund_operation.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        """Get the buy fee.
+
+        Returns:
+            dict: The buy fee information.
+
+        """
+        return {
+            "amount": self._model.fees.buy_fee.amount,
+            "currency": self._model.fees.buy_fee.currency,
+        }
```
Comment (alex-stone) [inline cdp/fund_operation.py:177]: Should this be a `FiatAmount` as well?

**Review by alex-stone — APPROVED** [review]
Looks good, there are some minor differences in the amount / types and rendering here, but we can always align those later

File: cdp/fund_operation.py (hunk 1 above)
Comment (rohan-agarwal-coinbase) [reply cdp/fund_operation.py:177]: will fix in follow-up PR

**Pull request merged**